File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0;
        int candidate = NULL;
        
        for (auto& num : nums) {
            if (count == 0) {
                candidate = num;
            }
            count += (num == candidate) ? 1 : -1;
        }
        return candidate;
    }
};

File: 169-majority-element/169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int ans = 0, number = 0;
        for (int num : nums) {
            if (ans == 0)
                number = num;
            if (number == num)
                ans++;
            else
                ans--;
        }
        return number;
    }
}